ksqldb-test-util/src/main/java/io/confluent/ksql/test/util/KsqlTestFolder.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.test.util;
+
+import java.io.File;
+import java.lang.ref.WeakReference;
+import org.junit.rules.TemporaryFolder;
+
+/**
+ * A Utility class to consolidate where we allocate state folders for tests.
+ */
+public final class KsqlTestFolder {
+
+  /**
+   * Uninstantiable static utility class
+   */
+  private KsqlTestFolder() { }
+
+
+  /**
+   * Create a temporary folder for testing. Rather than the system default
+   * temp directory, use the current working directory to avoid directories
+   * being deleted during tests.
+   */
+  public static TemporaryFolder temporaryFolder() {
+    final TemporaryFolder temporaryFolder = TemporaryFolder
+        .builder()
+        .parentFolder(new File(System.getProperty("user.dir")))
+        .build();
+
+    // Just in case the JVM exits without a proper test failure.
```

Comment:
https://stackoverflow.com/questions/20829228/shutdown-hook-vs-finalizer-method

I guess finalize is less reliable.

Reply:
Thanks! I actually hadn't thought of the finalizer, since this is more related to what happens when the JVM exits abruptly. For situations where the object would just get GCed normally, we'd expect JUnit to already have called `after()`.

But the finalizer might be a good way to try and remove the shutdown hook after the test finishes. I'll give that a shot before merging.

Reply:
Oh, nevermind. That would also require wrapping the TemporaryFolder with a new object, which I already tried (to intercept `after()` and unregister the hook). It can be done, but it's a pain.

I think I'll just merge this as-is for now.